src/installer.ts

```diff
@@ -77,20 +80,57 @@ async function acquireGo(version: string): Promise<string> {
     throw new Error('Temp directory not set');
   }
 
-  if (osPlat == 'win32') {
+  // gotip is always .tar.gz.
+  if (osPlat == 'win32' && !gotip) {
     extPath = await tc.extractZip(downloadPath);
   } else {
     extPath = await tc.extractTar(downloadPath);
   }
 
+  let toolRoot = extPath;
+  if (!gotip) {
+    toolRoot = path.join(toolRoot, 'go');
+    version = normalizeVersion(version);
+  }
+
+  //
+  // Build gotip from source.
+  //
+  if (gotip) {
+    // Version embed in the executable. Required for build.
+    let versionFile = path.join(toolRoot, 'VERSION');
+    fs.writeFileSync(versionFile, `gotip-${version}`);
```

Comment:
This isn't quite the same as what building directly from a git clone would output. On stable, the equivalent to this file would be `go1.13.5`. On tip, it'd be `devel +<hash>`, where `<hash>` is the shortened git commit hash produced by `git rev-parse --short HEAD`. Unfortunately, I'm not sure you can get the shortened string without actually cloning as truncating a hash isn't the same logic, as `--short` means "give me at least this many unique characters" to deal with potential collisions.

Reply:
Fixed.